### PR TITLE
Tidying up tests

### LINKED ISSUE #115: Tidy up test data
All `@pytest.fixtures` can be moved to `tests/conftest.py` so they are in one place.

Normally when I have files I use to test results against I place them under `tests/resources` so that they are contained within `tests` where they are used. At the moment files are under `data/`. Its not strictly necessary but helps keep things tidy just in case there are things under `data/` that are used elsewhere.  

If there are files that aren't part of tests provided in `data/` that need provisioning as  part of a packages "resources" this can be done in several different ways, see [this example repository](https://github.com/wimglenn/resources-example) for the different methods. #108 introduces using `pkgutil` to provision resources for the parameterisation (see issue #7 and #97 which did so incorrectly).

### PR DESCRIPTION
**Pull request by ns-rse**
Resolves #115 

Moving all fixtures in conftest.py and test data under tests/resources/data

### CODE REVIEW

**Comment by bobturneruk** [comment]
We have two mass lists here: https://github.com/Mesnage-Org/pgfinder/tree/master/data/masses One for *E. coli*, another for *C. diff*. Does this replace one, or both? Also the strings in the new version look quite different compare `gm(-Ac) (x2) |0` (old *C. diff*) and `gm-AE|1` (new). No (-Ac) or (x2) anywhere in the new version. And we used to just have `L` not `[[I/L]/L]`, I think. Maybe needs discussion.

**Comment by ns-rse** [comment]
@bobturneruk don't worry its not ready yet, I need to fix the paths to work with Windows (its failing because the test data use `/` but on Windows they come out as `\`). Looking at that this afternoon.

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
+"""Fixtures for various tests."""
+from pathlib import Path
+import pytest
+import pandas as pd
+
+import pgfinder.pgio as pgio
+import pgfinder.validation as validation
+
+DATA_DIR = Path("tests/resources/data")
```
Comment (bobturneruk) [inline tests/conftest.py:9]: This seems better than the previous hardcode.

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+# test_matching.py fixtures
+@pytest.fixture
```
Comment (bobturneruk) [inline tests/conftest.py:12]: Makes sense for stuff like this to be in `conftest.py`.

File: tests/resources/data/baseline_output_ftrs_10columns.csv [hunk 3]
```diff
@@ -0,0 +1,3987 @@
+file: data/ftrs_test_data.ftrs masses_file: data/masses/e_coli_monomer_masses.csv modifications: - Sodium - Potassium - Anhydro - DeAc - Deacetyl_Anhydro - Nude - Decay - Amidation - Amidase - Double_Anh - multimers_Glyco ppm: 10 rt_window: 0.5 ,ID,xicStart,xicEnd,ionCount,chargeOrder,rt,mwMonoisotopic,theo_mwMonoisotopic,inferredStructure,maxIntensity
```
Comment (bobturneruk) [inline tests/resources/data/baseline_output_ftrs_10columns.csv:1]: How / where is this file used?

File: tests/resources/data/parameters.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
+// Multimer type
```
Comment (bobturneruk) [inline tests/resources/data/parameters.json:1]: How / where is this file used, please?

**Review by bobturneruk — COMMENTED** [review]
Couple of questions about files that I can't figure out. Also, this introduces some duplication of files in the repo. I can see why we'd want the test data in the tests folder. Can we prune anything in the data folder? Should we be moving data into the package?

File: tests/resources/data/baseline_output_ftrs_10columns.csv (hunk 3 above)
Comment (ns-rse) [reply tests/resources/data/baseline_output_ftrs_10columns.csv]: Good spot, thanks. Its not used, now removed.

**Review by bobturneruk — APPROVED** [review]
Looks good!

**Pull request merged**